### ISSUE TITLE
Path decoding: more explicit

### ISSUE THREAD

**Issue by boxed** [TriOptima/iommi#250]
```py
register_path_decoding(
    invoice__pk=Invoice
 )
```

Deprecate old magic.

**Comment by boxed** [comment]
Test case for:

```register_path_decoding(user__contact__pk=User)` that results in `User.objects.get(contact__pk=....)`

**Comment by boxed** [comment]
New idea:

```py
register_path_decoding(
    invoice__name=Invoice.name,   # invoice__name can be any string you want
    invoice_pk=Invoice,  # short for Invoice.id (if 'id' is the name of the primary key column)
)
```

If you do `foo=Invoice.pk`, that is a `property` that has no information on where it came from. So we want a nice error message to tell the user that if they tried `to do .pk` then they can't do that but must use `.id` or just the model itself.

**Comment by boxed** [comment]
Fixed in #316
